portindex2json/portindex2json.tcl

```diff
@@ -12,6 +12,74 @@
 
 package require json::write
 
+proc format_strings {strings} {
+    set new_strings {}
+    foreach element $strings {
+        lappend new_strings [::json::write string $element]
+    }
+    return $new_strings
+}
+
+proc get_maintainer_array {maintainer} {
+    set detail_list [split $maintainer @]
+    if {[string first : [lindex $detail_list 0]] != -1} {
+        set email_list [split [lindex $detail_list 0] :]
+        set name [string trimright [lindex $email_list 1]]
+        set email(name) "\"$name\""
+        set email(domain) "\"[lindex $email_list 0]\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    } else {
+        set name [string trimright [lindex $detail_list 0]]
+        set email(name) "\"$name\""
+        set email(domain) "\"macports.org\""
+        ::json::write indented false
+        set email_subobject [::json::write object {*}[array get email]]
+        ::json::write indented true
+    }
+
+    set maintainer_array(github) "\"[lindex $detail_list 1]\""
+    set maintainer_array(email) $email_subobject
+
+    return [array get maintainer_array]
+}
+
+proc parse_maintainers {maintainers} {
+    set ismaintainer 0
+    foreach maintainer $maintainers {
+        if {$maintainer == "openmaintainer"} {
+            continue
+        } elseif {$maintainer == "nomaintainer"} {
+            continue
+        } else {
+            lappend maintainer_subobjects [::json::write object {*}[get_maintainer_array $maintainer]]
+            set ismaintainer 1
+        }
+    }
+    if {$ismaintainer == 1} {
+        set maintainers_list [::json::write array {*}$maintainer_subobjects]
+        unset maintainer_subobjects
+        return $maintainers_list
+    } else {
+        return "\[\]"
+    }
+}
+
+proc is_closedmaintainer {maintainers} {
+    set closedmaintainer true
+    foreach maintainer $maintainers {
+        if {$maintainer == "openmaintainer" || $maintainer == "nomaintainer"} {
+            set closedmaintainer false
```

Comment:
> If you have one million maintainers and the first one is `openmaintainer`, you would keep iterating through all of them, just to return `false` at the end anyway. Don't set the variable, just return false here.
> 

Thanks


> (I would somewhat prefer if the two functions (`is_closedmaintainer` and `parse_maintainers`) would be combined into one to avoid doing the same work twice, but it's true that it's then a tiny bit more convoluted code if one needs to return two results, fetch the first one, then the second one etc.)

We have to set values for two keys, so I could think of this approach only.